main/src/ca/uwaterloo/flix/language/errors/SafetyError.scala

```diff
@@ -14,6 +14,27 @@ sealed trait SafetyError extends CompilationMessage {
 
 object SafetyError {
 
+  /**
+    * An error raised to indicate an unexpected pattern bound in a body atom.
+    *
+    * @param loc the position of the body atom containing the unexpected pattern.
+    */
+  case class UnexpectedPatternInBodyAtom(loc: SourceLocation) extends SafetyError {
+    def summary: String = s"Unexpected pattern in body atom."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Unexpected pattern in body atom.
+         |
+         |${code(loc, "pattern occurs in this body atom.")}
+         |""".stripMargin
+    }
+
+    // TODO: Proper explain for this!
```

Comment:
We try not to leave TODOs in the code (preferring issues instead). Here the TODO is so small I would do neither, so just remove the comment.